cmd/argocd/commands/app.go

```diff
@@ -673,9 +673,10 @@ func NewApplicationUnsetCommand(clientOpts *argocdclient.ClientOptions) *cobra.C
 				}
 				for _, env := range pluginEnvs {
```

Comment:
This might be a breaking change for some users. For backward compatibility please add `--force` CLI flag. If it is set to true then unset should act idempotently.

Reply:
Per my comments in https://github.com/argoproj/argo-cd/issues/6325#issuecomment-859528827

> Reviewed related code. It turns out only unset --plugin-env will return errors on not found while the others do return OK.

> I think it is better to keep a consistent behavior. What do you think? @alexmt

I am considering current `unset APPNAME --plugin-env`'s behavior as a bug and needs to be corrected. Do we want to keep backward compatibility of such feature too?

There are also some other error handling in the `unset` command. Should the `--force` flag override all or just plugin env not found?

Reply:
Thanks for checking other flags behavior. I agree with you, perhaps `--plugin-env` validation was added by mistake.